server/routes/api.js

```diff
@@ -0,0 +1,61 @@
+const express = require("express");
+const router = express.Router();
+const request = require("request");
+const City = require(`../models/City`);
+
+const apiKey = `e1c1e3622a802b737a0b62a19ddced05`;
+const getWeatherUrl = function(query) {
+  if (query.cityName) {
+    const { cityName } = query;
+    return `https://api.openweathermap.org/data/2.5/weather?q=${cityName}&units=metric&appid=${apiKey}`;
+  } else {
+    const { lat, lon } = query;
+    return `https://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lon}&units=metric&appid=${apiKey}`;
+  }
+};
+
+router.get("/city", function(req, res) {
+  request(getWeatherUrl(req.query), function(error, response) {
+    if (response.statusCode == 200) {
+      const data = JSON.parse(response.body);
+      const releventData = {
+        name: data.name,
+        tempature: Math.round(data.main.temp),
+        condition: data.weather[0].description,
+        conditionPic: `http://openweathermap.org/img/wn/${data.weather[0].icon}@2x.png`
+      };
+      res.send(releventData);
+    } else {
+      res.send("error");
+    }
+  });
+});
+
+router.get(`/cities`, async (req, res) => {
+  let cities = await City.find({});
+  res.send(cities);
+});
+
+router.post(`/city`, async (req, res) => {
+  const newCity = new City(req.body);
+  if ((await City.find({ name: newCity.name })).length > 0) {
```

Comment:
You can check if the city exists by comparing it with the `req.body.name`, and if the city doesn't exist, create the newCity (there is no need to create newCity if the city already exists.